src/auth/ejabberd_auth.erl

```diff
@@ -316,6 +328,30 @@ does_user_exist(#jid{luser = LUser, lserver = LServer}) ->
 does_user_exist(error) ->
     false.
 
+%% Hook interface
+-spec does_user_exist(HostType, Jid, RequestType) -> Res when
+      HostType :: mongooseim:host_type(),
+      Jid :: jid:jid(),
+      RequestType :: stored | with_anonymous,
+      Res :: boolean().
+does_user_exist(HostType, Jid, RequestType) ->
+    mongoose_hooks:does_user_exist(HostType, Jid, RequestType).
+
+%% Hook handler
+-spec does_user_exist(Status :: boolean(),
+                      HostType :: mongooseim:host_type(),
+                      Jid :: jid:jid(),
+                      RequestType :: stored | with_anonymous) -> boolean().
+does_user_exist(false, HostType, Jid, stored) ->
+    does_stored_user_exist(HostType, Jid);
```

Comment:
`does_user_exist` is supposed to return a boolean, but it calls a function that can return `{error, any()}`. In this case we should treat errors as `false` and add a comment that says so to the function header.

```suggestion
    true =:= does_stored_user_exist(HostType, Jid);
```